cumulusci/robotframework/pageobjects/PageObjects.py

```diff
@@ -254,6 +267,46 @@ def load_page_object(self, page_type, object_name=None):
         self._set_current_page_object(pobj)
         return pobj
 
+    @capture_screenshot_on_error
+    def wait_for_dialog(self, page_type, object_name, expected_heading=None, **kwargs):
+        """Wait for the given page object dialog to appear.
+
+        This will both wait for the dialog, and verify that the
+        dialog has an expected heading. The expected heading will be
+        the page type (eg "New") and object name (eg: "Contact")
+        separated by a space (eg: "New Contact").
```

Comment:
This doesn't quite match the implementation--from the docstring it sounds like passing page_type=Edit and object_name=Contact would use an expected heading of "Edit Contact"

Reply:
Also it seems like bad coupling that the base PageObjects class needs to know that a type of page object for dialogs exists. Why do we need a specific `wait_for_dialog` instead of just using the more generic `wait_for_page_object`?

Reply:
I mainly kept the two for readability in tests. I think a test is more readable with `wait for dialog` (or `wait for modal` than `wait for page object` since that's what you would ask a manual tester to do. 

OTOH, there will be times when a page object isn't a modal. A page object could be something else such as a complex lightning component. In that case, `wait for modal` would be wrong since it's not waiting on a modal.

Reply:
That mostly has me convinced. Readability in tests is important. On the other hand, `wait for page object` gives the reader a better clue that the test is switching to a different page object.

Reply:
I think in the interest of moving forward I'm going to keep both of these keywords, unless you feel strongly that we should only have one.